thrift/arg2/kv_iterator.go

```diff
@@ -53,32 +58,34 @@ func (i KeyValIterator) Next() (KeyValIterator, error) {
 		return KeyValIterator{}, io.EOF
 	}
 
-	arg2Len := len(i.arg2Payload)
-	cur := i.valueOffset + i.valueLen
-	if cur+2 > arg2Len {
-		return KeyValIterator{}, fmt.Errorf("invalid key offset %v (arg2 len %v)", cur, arg2Len)
+	if i.rbuf.BytesRemaining() < 2 {
```

Comment:
previously, we had a bunch of error checks to avoid panics, although now with readbuf, since it stores errors internally, we can consider only checking errors once at the end if the `rbuf.Err() != nil`, what do you think?

Reply:
i considered that, but the errors returned by `rbuf.Err()` lacks helpful details for surfacing the issue, so decided it was better to maintain the current set of error checks. what we could do though, is to do a `rbuf.Err()` check at the end, *in addition* to the current set of checks to avoid missing any edge cases.

Reply:
I'm OK with doing both checks, or just checking `rbuf.Err()`. Agree that it doesn't provide many details, but given that we don't expect malformed frames, and providing more details on how the frame is malformed likely won't help clients anyway, it doesn't seem too important to provide that the more granular errors.

Reply:
actually looking at the code a bit more, the final check might not be necessary, as the existing checks already cover possible failures. i vote for leaving the way it currently is.

Reply:
oops didn't see your comments before posting this. in that case let's just consistently do a single check at the end.